### PR TITLE
[IWJ-15] display actual data and make transfer form available

### PR DESCRIPTION
**Pull request by oimou**
# Ticket
https://soramitsu.atlassian.net/browse/IWJ-15

# Description
Main updates are below:

1. Create a Vuex store module `Account.js` which encapsulates all Iroha connections.
2. Using Account.js, display actual data and make the transfer form available.
screen shot 2018-05-16 at 17 28 31

3. Show background connection errors with [ElementUI's message component](https://element.eleme.io/#/en-US/component/message):
screen shot 2018-05-16 at 17 16 40

Note: since currently the server hangs sometimes, you might need to re-create accounts/assets/transactions in DB for each time. `example/setup-accounts-and-assets.js` comes in for that case.

# How to check
1. `yarn install` to install new dependencies.
2. `yarn rebuild` to rebuild/reinstall native modules (grpc, iroha-lib) of electron version.
    * if you got an error about iroha-lib, you should put [iroha-lib binary](https://github.com/hyperledger/iroha-javascript/releases/tag/0.0.6) to `node_modules/iroha-lib/lib` manually.
3. `yarn dev` to open the app.

### CODE REVIEW

**Comment by oimou** [comment]
Update: This getAccountTransactions's behaviour is intended.

---

#### description
I tried to get account's all transfer transactions via getAccountTransactions in order to display on SummaryPage. However, I found that a sender (e.g. admin@test) can see his transfer transactions to other accounts, but a recipient (e.g. alice@test) cannot see that transfer transactions.

I wrote `example/alice-transactions-iroha-lib.js` and tested [getAccountAssetTransactions](https://hyperledger.github.io/iroha-api/?protobuf#get-account-asset-transactions) and [getAccountTransactions](https://hyperledger.github.io/iroha-api/?protobuf#get-account-transactions).

As a result, it seems
* getAccountAssetTransactions can get transactions Alice is recipient, but
* getAccountTransactions cannot get such transactions.

Maybe my understanding about getAccountTransactions is wrong? 🤔 

#### logs

<details>
<summary>example/alice-transactions-iroha-lib.js</summary>

```
[~/work/soramitsu/iroha-wallet-js] (feature/display-correct-data|✚1…)
 % node example/alice-transactions-iroha-lib.js
accountId: alice@test
assetId: coolcoin#test
nodeIp: 51.15.244.195:50051

getAccountAssetTransactions: {
  "transactionsResponse": {
    "transactionsList": [
      {
        "payload": {
          "commandsList": [
            {
              "transferAsset": {
                "srcAccountId": "admin@test",
                "destAccountId": "alice@test",
                "assetId": "coolcoin#test",
                "description": "",
                "amount": {
                  "value": {
                    "first": 0,
                    "second": 0,
                    "third": 0,
                    "fourth": 10000
                  },
                  "precision": 2
                }
              }
            }
          ],
          "creatorAccountId": "admin@test",
          "txCounter": 1,
          "createdTime": 1525939973731
        },
        "signatureList": [
          {
            "pubkey": "QH5X9QykiWmwi6lIFxuyQ14DXYLOxBfhjko49fsRP4M=",
            "signature": "QHKL54zkSmMiJyVuSQiNKCZ0phaeHTM7Azmao3yWOqYILGLekDgMpsoDeSIkcorRnrMRZ2ZrkEiZywNTcpDKDw=="
          }
        ]
      }
    ]
  },
  "queryHash": "pzmFRaSekwCmr3nruED3a05LXIGcu2xefcABLdrGNDU="
}

getAccountTransactions: {
  "transactionsResponse": {
    "transactionsList": []
  },
  "queryHash": "9Q6Poqjes2GTzZhlPTrwPh1NOg8Ve2yka/vnO75NhpM="
}
```
</details>

<details>
<summary>iroha-cli (just in case)</summary>

```
root@3b7abe3a7fae:/opt/iroha_data# iroha-cli -account_name alice@test -peer_ip 51.15.244.195
Welcome to Iroha-Cli.
Choose what to do:
1. New transaction (tx)
2. New query (qry)
3. New transaction status request (st)
> : qry
Choose query:
1. Get all permissions related to role (get_role_perm)
2. Get Transactions by transactions' hashes (get_tx)
3. Get information about asset (get_ast_info)
4. Get Account's Transactions (get_acc_tx)
5. Get all current roles in the system (get_roles)
6. Get Account's Signatories (get_acc_sign)
7. Get Account's Assets (get_acc_ast)
8. Get Account Information (get_acc)
0. Back (b)
> : get_acc_ast
Requested account Id: alice@test
Requested asset Id: coolcoin#test
Query is formed. Choose what to do:
1. Send to Iroha peer (send)
2. Save as json file (save)
0. Back (b)
> : send
Peer address (51.15.244.195): 51.15.244.195
Peer port (50051): 50051
[2018-05-11 05:16:54.178985200][th:11498][info] QueryResponseHandler [Account Assets]
[2018-05-11 05:16:54.179112000][th:11498][info] QueryResponseHandler -Account Id:- alice@test
[2018-05-11 05:16:54.179131700][th:11498][info] QueryResponseHandler -Asset Id- coolcoin#test
[2018-05-11 05:16:54.179265000][th:11498][info] QueryResponseHandler -Balance- 100.00
--------------------
Choose what to do:
1. New transaction (tx)
2. New query (qry)
3. New transaction status request (st)
> : qry
Choose query:
1. Get all permissions related to role (get_role_perm)
2. Get Transactions by transactions' hashes (get_tx)
3. Get information about asset (get_ast_info)
4. Get Account's Transactions (get_acc_tx)
5. Get all current roles in the system (get_roles)
6. Get Account's Signatories (get_acc_sign)
7. Get Account's Assets (get_acc_ast)
8. Get Account Information (get_acc)
0. Back (b)
> : get_acc_tx
Requested account Id: alice@test
Query is formed. Choose what to do:
1. Send to Iroha peer (send)
2. Save as json file (save)
0. Back (b)
> : send
Peer address (51.15.244.195): 51.15.244.195
Peer port (50051): 50051
--------------------
Choose what to do:
1. New transaction (tx)
2. New query (qry)
3. New transaction status request (st)
> :
```
</details>

File: src/renderer/store/modules/Account.js [hunk 1]
```diff
@@ -0,0 +1,256 @@
+import _ from 'lodash'
+import grpc from 'grpc'
+import irohaUtil from 'util/iroha-util'
+
+// TODO: get assetIds via API in the future
+const DUMMY_ASSET_IDS = [
```
Comment (oimou) [inline src/renderer/store/modules/Account.js]: Dummy assets are hardcoded here.

**Comment by oimou** [comment]
* [x] 1) I think when you send a transaction over APP you should check for "STATELESS_VALIDATION_SUCCESS" first, and after it maybe send a hash to iroha to check if transaction is successfull
  * -> cannot reproduce
* [x] 2) Please, change the data format to the folloving: Apr. 9, 12:34 (so it is easier to check transactions, so they all are at the same day).
* [x] 3). For some reason, I have duplicates of a transaction in the list. I've created only 2 transfer in the Iroha-cli, yet they duplicate in the list. Please investigate it.
* [x] remove "id" field

**Comment by laSinteZ** [comment]
* [x] If you delete `node-modules` and write `yarn`, this happens: (I have nodejs 10)
error upath@1.0.4: The engine "node" is incompatible with this module. Expected version ">=4 <=9". 

fixed it manualy in yarn.lock 
```
upath@^1.1.0:
  version "1.1.0"
  resolved "https://registry.yarnpkg.com/upath/-/upath1.1.0.tgz#35256597e46a581db4793d0ce47fa9aebfc9fabd"
```

 * [x] don't forget to updage package.json and include yourself as an author, since there is my name for some reason, and I mostly did nothing :D

* [x] (side panel should obviously not scroll to bottom. though it is not realted to this pr please fix it)🙏 It has a scroll if there are too many transactions in history
[screen shot 2018-05-17 at 17 12 55
]()

I guess that's all for now. Issue with upath may be resolved if you successfully migrate to vue-cli 3 :) Overall LGTM, after fixing those you can merge and proceed to next task :)

**Pull request merged**